scrapy_autoextract/providers.py

```diff
@@ -0,0 +1,61 @@
+from typing import ClassVar, Optional, Type
+
+from autoextract.aio import request_raw
+from autoextract.request import Request as AutoExtractRequest
+from autoextract_poet.page_inputs import (
+    AutoExtractArticleData,
+    AutoExtractProductData,
+)
+from scrapy import Request
+from scrapy.settings import Settings
+from scrapy.statscollectors import StatsCollector
+from scrapy_poet.page_input_providers import PageObjectInputProvider
+
+
+class _Provider(PageObjectInputProvider):
+
+    page_type: ClassVar[str]
+    provided_class: ClassVar[Optional[Type]]
```

Comment:
would it be possible to use the current scrapy-poet providers API?

Reply:
Done. Keep in mind we need to import scrapy-autoextract's providers module in order to effectively import its providers.

Reply:
There is no provided_class in the current scrapy-poet - how does it work?

Regarding having to import it, what do you think about not using `@provides` decorator, but creating a helper which registers all available providers?

```py
from scrapy_poet.page_input_providers import register

def install():
    register(ArticleResponseDataProvider, AutoExtractArticleData)
    register(ProductResponseDataProvider, AutoExtractProductData)
```

And then document that one should put the following code to settings.py to be able to use scrapy-autoextract with page objects?

```py
import scrapy_autoextract.poet
scrapy_autoextract.poet.install()
```

Reply:
I had the same feeling about this decorator. I guess it will leave on the next versions. Let me try to implement something based on your suggestion.

Reply:
Done.